var/spack/repos/builtin/packages/camp/package.py

```diff
@@ -55,6 +55,8 @@ class Camp(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on("blt", type="build")
 
+    patch("gcc-13.patch", when="@:2022.10.1 %gcc@13")
```

Comment:
For the change in `include/camp/resource.hpp` it would probably make sense to use the patch from https://github.com/LLNL/camp/pull/125/commits/05e1c35a5b393c0ab6139a03fb6bfd00c477568a (i.e. `https://github.com/LLNL/camp/pull/125/commits/05e1c35a5b393c0ab6139a03fb6bfd00c477568a.patch?full_index=true`). Since this was added in 2023.06 the version constraint would be `@:2022.10` (again not sure about a lower bound. Would you mind testing if older versions compile with the patch?)

The change to `include/camp/resource.hpp` isn't upstreamed but the code has changed not to be an issue from 2023.06 onwards so keeping `gcc-13.patch` for that seems fair. Perhaps you can rename it `gcc-13-runtime-error-cstdlib.patch` or similar?

Also `13:` for the GCC version and the patch version can be removed from `2022.10.1`:
```suggestion
    patch("gcc-13.patch", when="@:2022.10 %gcc@13:")
```

Reply:
OK, your suggestion is fair. I would rename the patch, though testing older versions of camp maybe a little hard: actually these headers are copied straightforward without a compile. It needs compile some other package using camp headers, e. g. umpire, to test if the patch works well.
Besides, It seems the patch is compatible in pattern at least with the 2020 "First numbered release"(https://github.com/LLNL/camp/releases/tag/v0.1.0), I think it would not be a big problem using @ :2022.10 .